libpod/events/journal_linux.go

```diff
@@ -97,8 +98,9 @@ func (e EventJournalD) Write(ee Event) error {
 }
 
 // Read reads events from the journal and sends qualified events to the event channel
-func (e EventJournalD) Read(ctx context.Context, options ReadOptions) error {
-	defer close(options.EventChannel)
+func (e EventJournalD) Read(ctx context.Context, options ReadOptions) (retErr error) {
+	runtime.LockOSThread()
```

Comment:
Is it obvious why this is necessary? If not, a comment would be nice.

Reply:
Oh yeah I think I can drop this gain. It is not doing anything AFAICT especially since we span a new goroutine in here anyway where we call from a different thread then. I think I added this because I had same badfd/closed fd errors that were caused by a bug that closed the journal fd to early but I fixed that

Reply:
(I’m not at all saying that it’s unnecessary; just that, reading the PR, I don’t know what to look for.)

Reply:
Yes I got that by I remembered why I added it and that was chasing a ghost that turned out to be a simple close to early when I was debugging locally.
I agree that it is confusing as it is not clear to readers why this is here so it should just be dropped. I will do so tomorrow.